### PR TITLE
[external assets] move api endpoint code & tests

### PR DESCRIPTION
**Pull request by alangenfeld**
to better group these for coming other endpoints

## How I Tested These Changes

bk

### CODE REVIEW

**Comment by alangenfeld** [comment]
Current dependencies on/for this PR:
* master
  * **PR #17133** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/17133" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/17133?utm_source=stack-comment).

File: python_modules/dagster-webserver/dagster_webserver_tests/webserver/test_asset_events.py [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+            ), "need to add validation that sample payload content was written successfully"
+
+    # all ext report_asset_materialization kwargs should be in sample payload
+    sig = inspect.signature(PipesContext.report_asset_materialization)
+    skip_set = {"self"}
+    params = [p for p in sig.parameters if p not in skip_set]
+
+    KNOWN_DIFF = {"partition", "description"}
+
+    assert set(sample_payload.keys()).difference(set(params)) == KNOWN_DIFF
```
Comment (alangenfeld) [inline python_modules/dagster-webserver/dagster_webserver_tests/webserver/test_asset_events.py:180]: i'm not sure about aligning the rest endpoint to pipes context methods. They behave differently:
* pipes `report` calls --> result objects in the orchestration process --> asset events from the result objects in system code
* this endpoint --> asset events directly

This difference is why `partition` is not supported in pipes but is for the direct report in the endpoint (usually taken care of by system code)

I guess the overall goal is to ensure any future arguments that translate directly from result object on to underlying event get caught and supported in both places.

Comment (schrockn) [reply python_modules/dagster-webserver/dagster_webserver_tests/webserver/test_asset_events.py:180]: Yeah we don't have to add it know.

Where do you think is the right spot in the system to ensure parity between APIs in the main stack and the REST APIs?

**Review by schrockn — APPROVED** [review]
[didntreaddance](https://app.graphite.dev/macros/dagster-io)

**Pull request merged**